### ISSUE TITLE
Best Practices for Fine-Tuning CellPose SAM on Custom Datasets

### ISSUE THREAD

**Issue by vadori** [MouseLand/cellpose#1235]
Hi everyone,

I'm seeking advice on fine-tuning the pre-trained CellPose SAM model with custom data. I'd be grateful for any insights on the following points:

- Is it more effective to fine-tune the entire model or only the final layer(s)?
- Are the default learning rate settings generally suitable, or would adjustments be recommended?
- How many annotated images (or individual cells) are typically needed to fine-tune effectively while minimizing the risk of overfitting?

Thank you in advance for your suggestions!

**Comment by mrariden** [comment]
If you use our training code, the defaults are set for most use cases. We can't say how many images/cells are needed because it depends on how similar your cells are to our training data. If you use our training code, you don't need to worry about overfitting because we pretrain on a generalist dataset. 

More info is available on the read the docs page [here](https://cellpose.readthedocs.io/en/latest/faq.html) and [here](https://cellpose.readthedocs.io/en/latest/train.html)

**Comment by vadori** [comment]
Great, thanks @mrariden. I’ll run some tests. Regarding overfitting, I understand the risk is lower compared to using models not trained on such extensive datasets. However, I wouldn’t entirely dismiss the possibility. It could still occur to some extent if the weights adapt too much to my custom dataset and “forget” the original training data.
